recipes/cppwinrt/all/conanfile.py

```diff
@@ -0,0 +1,108 @@
+import os
+
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get
+from conan.tools.microsoft import MSBuild, MSBuildDeps, MSBuildToolchain, VCVars, check_min_vs, is_msvc, msbuild, vs_layout
+
+required_conan_version = ">=1.53.0"
+
+
+class CppWinRtConan(ConanFile):
+    name = "cppwinrt"
+    description = "C++17 language projection for Windows Runtime"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/microsoft/cppwinrt"
+    topics = ("native", "cpp", "winrt")
+    settings = "os", "arch", "compiler", "build_type"
+
+    def layout(self):
+        vs_layout(self)
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def requirements(self):
+        self.requires("winmd/1.0.210629.2")
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    def validate(self):
+        if self.info.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        check_min_vs(self, 192)
+        if not is_msvc(self):
+            raise ConanInvalidConfiguration(
+                f"{self.ref} can be built only by Visual Studio and msvc.")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        self.conf.define("tools.microsoft.msbuild:verbosity", "Diagnostic")
```

Comment:
I think that we should not do this. Any consumer can adjust the verbosity value from their profiles. Still, if you want consumers to have that configuration automatically, then we should remove this line and add the following one in `package_info()`:

```python
    def package_info(self):
        self.conf_info.define("tools.microsoft.msbuild:verbosity", "Diagnostic")
        # ...
```

We should only read/check configuration values in the `generate()` function.

Reply:
That was just for debugging. Good catch.